surefire-booter/src/main/java/org/apache/maven/surefire/booter/ForkedBooter.java

```diff
@@ -129,6 +134,34 @@ private void setupBooter( String tmpDir, String dumpFileName, String surefirePro
 
         flushEventChannelOnExit();
 
+        final Long outputFlushInterval = providerConfiguration.getOutputFlushInterval();
+        if ( outputFlushInterval != null && outputFlushInterval > 0
+            && LegacyMasterProcessChannelEncoder.class.isInstance( eventChannel ) )
+        {
+            final LegacyMasterProcessChannelEncoder enc = LegacyMasterProcessChannelEncoder.class.cast( eventChannel );
+            flusher = Executors.newSingleThreadScheduledExecutor( DaemonThreadFactory.newDaemonThreadFactory() );
+            flusher.scheduleAtFixedRate( new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    final WritableBufferedByteChannel c = WritableBufferedByteChannel.class.cast( enc.getOut() );
+                    try
+                    {
+                        if ( c.isOpen() )
```

Comment:
what about adding a 
`MasterProcessChannelEncoder#flush() `
method

this way we can get rid of all of the casts.
LegacyMasterProcessChannelEncoder will implement it this way, other implementations will not implement it or implement it as an empty method

Reply:
this is a SPI (so user facing) so didn't want to modify it since this is really a workaround :s